core/expect.ts

```diff
@@ -185,15 +185,27 @@ export class Matcher {
    }
 
    /**
-    * Checks that an array is empty
+    * Checks that an array is empty, a string is empty, or an object literal has no properties
     */
    public toBeEmpty() {
-      if (!Array.isArray(this._actualValue)) {
-         throw new TypeError("toBeEmpty requires value passed in to Expect to be an array");
+      if (null === this.actualValue || undefined === this.actualValue) {
+         throw new TypeError("toBeEmpty requires value passed in to Expect not to be null or undefined");
       }
 
-      if ((this.actualValue.length === 0) !== this.shouldMatch) {
-         throw new EmptyMatchError(this._actualValue, this.shouldMatch);
+      if (typeof this.actualValue === "string") {
```

Comment:
Very true, but I'm not sure in this case there is likely to be any difference in how we handle this in the future so I'm happy for them both to be in the same branch :)